### PR TITLE
Added support for CoinGecko's API

### PR DESCRIPTION
**Pull request by helloitsm3**
This PR contains support for CoinGecko's API.

Current API endpoints:

- Ping 
- Simple
- Coins
- Contract
- Asset Platforms
- Categories
- Exchanges
- Index
- Derivatives
- Exchange Rates
- Sesarch
- Trending
- Global
- Companies

@royderks do take a look at this.

### CODE REVIEW

File: with-coingecko/README.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# StepZen Example: `with-firebase`
```
Comment (royderks) [inline with-coingecko/README.md:1]: Could you update the title?

File: with-coingecko/stepzen.config.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+  "endpoint": "api/loopy-alpaca"
```
Comment (royderks) [inline with-coingecko/stepzen.config.json:2]: Could you rename this to match the endpoint in the README?

File: with-coingecko/index.graphql [hunk 3]
```diff
@@ -0,0 +1,21 @@
+schema
+  @sdl(
+    files: [
+      "types/assets.graphql"
```
Comment (royderks) [inline with-coingecko/index.graphql:4]: typically we put all the generated `.graphql` files in a folder named after the data source. So `types` would be renamed to `coingecko`. I'll add this to the contributing guide

File: with-coingecko/README.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+## Introduction
+
+This project builds a GraphQL API connected to CoinGecko's API using StepZen. We will use the `@rest` directive to generate our GraphQL schema.
```
Comment (royderks) [inline with-coingecko/README.md:5]: It would be nice to have a link to the Coingecko API docs here

**Review by royderks — CHANGES_REQUESTED** [review]
Overall looks good, thank you. Are there some interesting combinations between Coingecko REST endpoints that can be made using `@materializer`?

**Comment by helloitsm3** [comment]
> Overall looks good, thank you. Are there some interesting combinations between Coingecko REST endpoints that can be made using `@materializer`?

I'll look into `@materializer` and see if I can use that

**Review by royderks — APPROVED** [review]

**Pull request merged**